### ISSUE TITLE
Decrypt error when trying to split file

### ISSUE THREAD

**Issue by jayachithra** [py-pdf/PyPDF2#613]
I am trying to split a pdf into its pages. For few documents I get the error: **PdfReadError: File has not been decrypted**
![image](https://user-images.githubusercontent.com/23461501/113410134-fcd1c600-93b2-11eb-9a47-ddbcc1436c06.png)

I tried the following fixes, but nothing works.
Fix 1: Manually decrypt the file using: inputpdf.decrypt('')
This throws an NotImplementedError:
![image](https://user-images.githubusercontent.com/23461501/113410466-bc267c80-93b3-11eb-8b95-45fa0f0348c1.png)

Fix 2: Suggested here: https://stackoverflow.com/a/48364988/15368105
This also doesn't work and throws the same error. 

Is there a solution for this?

**Comment by averagechris** [comment]
Looks like this library is only able to decode 40 and 128 bit instead of more modern schemes.

I ran into this because an old library that depends on `pdftk` to encrypt pdfs started failing with the `NotImplementedError: only algorithm code 1 and 2 are supported` when trying to decrypt those pdfs.

Well if you're on ubutu or debian, your package manager will install a newer version of `pdftk` (the java version) by default now. And the newer version of pdftk defaults to an encryption scheme that this library doesn't support.

I was able to work around this temporarily (until moving to some other dependencies) by explicitly telling pdftk to use the 128 bit encryption instead.

e.g. `pdftk input.pdf output output.pdf encrypt_128bit owner_pw ....`

Hopefully this helps you figure out your issue if you find this same issue like I did!
